packages/fxa-payments-server/src/routes/Checkout/index.tsx

```diff
@@ -197,7 +197,8 @@ export const Checkout = ({
         } else {
           // Some Stripe APIs like `createPaymentMethod` return an object with
           // an error property on error.
-          setSubscriptionError(error?.error || error);
+          const err = typeof error.error === 'string' ? error : error.error;
+          setSubscriptionError(err);
```

Comment:
Now that I look back at why I added this, https://github.com/mozilla/fxa/pull/10127, we should just be able to pass in `error` without this ternary at all.

We already [destructure the Stripe API response (e.g. `PaymentMethodResult`) and pass in the value of its `error` property when defined](https://github.com/mozilla/fxa/blob/2cb4e855d3d52417b6b848b975302c1e48a06b42/packages/fxa-payments-server/src/lib/stripe.ts#L136-L143), which based on Stripe's types, is always an object. I've checked everywhere we're making a Stripe API call on the payments server.

Reply:
I did try just passing in the whole error object (`setSubscriptionError(error)` but it caused a test to fail ("displays an error when payment failed"). I did this to fix the currency mismatch error message and keep the existing tests passing, happy to simiplify this if the test that failed is no longer a case we need to worry about; just let me know (also pinged you on slack in case that's quicker but wanted to drop this here for record keeping purposes).

Reply:
Sure let's file a follow up. Thank you.

Reply:
I think the test is wrong, not the code FWIW. I think the Stripe API resolves even when there's an error, but the test is mocking it to reject, which is what's sending us into that `catch` block.